### PR TITLE
chore: rename navigation delegates

### PR DESCRIPTION
**Pull request by asalzburger**
Thi PR sits on top of #3194 and cleans up naming and file location, in fact - it overwrites some of the changes.

`NavigationDelegates.hpp`
- contains definition of the delegates and the internal and external navigation interfaces

`NavigationStateFillers.hpp`
- contains the fillers into the navigation state which can be used to specify the updators

`NavigationStateUpdaters.hpp`
- contains the generic updaters code that can be used to implement `IInternalNavigation` and `IExternalNavigation`

`InternalNavigation.hpp` 
- a standard set of internal navigation implementations, many of them using the navigation state updators

`MultiWireNavigation.hpp`
- a dedciated implementation of internal navigation for multi wire chambers

`PortalNavigation.hpp`
- single and multiple connected portal navigation implementations

`DetectorVolumeFinders.hpp`
- global volume finder implementations

Individual `....Navigation` classes implement either the internal or the external navigation interface.

### CODE REVIEW

File: Tests/UnitTests/Core/Navigation/PortalNavigationTests.cpp [hunk 1]
```diff
@@ -60,12 +60,12 @@ BOOST_AUTO_TEST_CASE(SingleVolumeUpdate) {
...
   BOOST_CHECK_EQUAL(nState.currentVolume, volumeA.get());
 
-  Acts::Experimental::SingleDetectorVolumeImpl svu(volumeB.get());
+  Acts::Experimental::SingleDetectorVolumeNavigation svu(volumeB.get());
   svu.update(tContext, nState);
 
   BOOST_CHECK_EQUAL(nState.currentVolume, volumeB.get());
 
-  BOOST_CHECK_THROW(Acts::Experimental::SingleDetectorVolumeImpl(nullptr),
+  BOOST_CHECK_THROW(Acts::Experimental::SingleDetectorVolumeNavigation(nullptr),
```
Comment (dimitra97) [inline Tests/UnitTests/Core/Navigation/PortalNavigationTests.cpp:68]: I think the exception does not work here, the test is failing

**Review by andiwand — APPROVED** [review]
lgtm

File: Tests/UnitTests/Core/Navigation/PortalNavigationTests.cpp (hunk 1 above)
Comment (asalzburger) [reply Tests/UnitTests/Core/Navigation/PortalNavigationTests.cpp:68]: Got it.

**Comment by acts-project-service** [comment]
## :red_circle: Athena integration test results [eeb854aa6061211e02c338e1d7cb87cc225e58b0]
### Build job with this PR failed!
Please investigate the [build job](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/39132085) for the [pipeline](https://gitlab.cern.ch/acts/acts-athena-ci/-/pipelines/7420021)!

**Comment by andiwand** [comment]
I think there is missing include which fires in Athena @asalzburger

**Pull request merged**